plugins/pi4j-plugin-ffm/src/main/java/com/pi4j/plugin/ffm/common/i2c/rdwr/I2CMessage.java

```diff
@@ -98,23 +102,23 @@ public I2CMessage from(MemorySegment buffer) throws Throwable {
         var address = (int) VH_ADDRESS.get(buffer, 0L);
         var flags = (int) VH_FLAGS.get(buffer, 0L);
         var len = (int) VH_LEN.get(buffer, 0L);
-        var bufMemorySegment = invokeExact(MH_BUF, buffer);
-        var buf = new byte[len];
-        for (int i = 0; i < buf.length; i++) {
-            buf[i] = bufMemorySegment.getAtIndex(ValueLayout.JAVA_BYTE, i);
-        }
-        return new I2CMessage(address, flags, len, buf);
+
+        var bufferSegment = (MemorySegment)VH_BUFFER.get(buffer, 0L);
+        var bytes = new byte[len];
+        bufferSegment.reinterpret(len).asByteBuffer().get(bytes, 0, len);
```

Comment:
@IAmNickNack In general it is not safe to get `ByteBuffer` from `MemorySegment` - it is not guaranteed to be backed (see https://download.java.net/java/early_access/loom/docs/api/java.base/java/lang/foreign/MemorySegment.html#asByteBuffer() in exception description).

The other thing, is that you are creating an intermediate object in heap, it is better to stick with `MemorySegment` methods in this classes.
Can you please fix this?